plugins/custom-class-name/prism-custom-class-name.js

```diff
@@ -0,0 +1,29 @@
+(function(){
+
+if (
+	typeof self !== 'undefined' && !self.Prism ||
+	typeof global !== 'undefined' && !global.Prism
+) {
```

Comment:
I think you have this wrong. What if both `self` and `global` are undefined? How about this:
```js
(typeof self === 'undefined' || !self.Prism) && (typeof global === 'undefined' || !global.Prism)
```